### PR TITLE
[needs-docs] Fix multiple raster calc issues (+ new operators)

### LINKED ISSUE #32023: Raster calculator change sign does not work when OpenCL is on
The title says it all

### LINKED ISSUE #32025: QGIS Raster Calculator outputs nodata only rasters
**Describe the bug**
When native qgis raster calculators are used in a model, they (sometimes) generate output rasters containing only nodata values. This does not happen when doing the same calculation using a processing toolbox raster calculator (outside the model).

**How to Reproduce**
Has already been reproduced by Alessandro. :)

**QGIS and OS versions**
3.9 master

### PR DESCRIPTION
**Pull request by elpaso**
Fixes #32023 Raster calculator change sign does not work when OpenCL is on
Fixes #32025 QGIS Raster Calculator outputs nodata only rasters

Bonus: three new operators with full test coverage
- ABS
- MIN
- MAX

I hope it is not a problem if I sneaked in three new operators, I've fixed the bugs while I was developing the operators, they have full test coverage so I'm pretty confident I'll not break anything.

### CODE REVIEW

**Comment by DelazJ** [comment]
>Bonus: three new operators with full test coverage

missing [needs-docs]?

File: src/analysis/raster/qgsrastermatrix.cpp [hunk 1]
```diff
@@ -18,6 +18,7 @@
 #include "qgsrastermatrix.h"
 #include <cstring>
 #include <cmath>
+#include<algorithm>
```
Comment (nyalldawson) [inline src/analysis/raster/qgsrastermatrix.cpp]:
```suggestion
#include <algorithm>
```

File: tests/src/analysis/testqgsrastercalculator.cpp [hunk 2]
```diff
@@ -587,6 +580,18 @@ void TestQgsRasterCalculator::findNodes()
   QVERIFY( ! node );
   QVERIFY( ! errorString.isEmpty() );
 
+  // Test new abs, min, max
+  errorString = QString();
```
Comment (nyalldawson) [inline tests/src/analysis/testqgsrastercalculator.cpp]:
```suggestion
  errorString.clear();
```

**Review by nyalldawson — APPROVED** [review]

**Comment by elpaso** [comment]
@nyalldawson thank you for the review!

**Pull request merged**